oppia/oppia-android
# Fix #1736: Add Bazel Linter to the project

## Explanation

Fix #1736 

This is the first part where we want to add linter for bazel. 

### How to test
1. Run command `brew install buildifier` on your root - this will install the buildifier
2. Go to oppia-android project root
3. You can either run command for single files or all at once
```
 buildifier --lint=warn --mode=check --warnings=all oppia_android_test.bzl
 buildifier --lint=warn --mode=check --warnings=all BUILD.bazel
 buildifier --lint=warn --mode=check --warnings=all WORKSPACE
or 
 buildifier --lint=warn --mode=check --warnings=all oppia_android_test.bzl BUILD.bazel WORKSPACE
```
4. If you want to see what auto-fix of this linter do
```
 buildifier --lint=fix --warnings=all oppia_android_test.bzl
 buildifier --lint=fix --warnings=all BUILD.bazel
 buildifier --lint=fix --warnings=all WORKSPACE
or 
 buildifier --lint=warn --mode=check --warnings=all oppia_android_test.bzl BUILD.bazel WORKSPACE
```

There are currently two types of issue which auto-fix cannot fix
1. out-of-order-load - lexicographical order, mentioned in the comments below
2. unsorted-dict-items - 

### How to review
Until now in this PR, I had fixed and added only root bazel files in checks. Before moving forward with the bazel files in modules, we need to clarify a few points which are a bit different way a linter fix the issues.

Now
```
./buildifier --lint=warn --mode=check --warnings=all BUILD.bazel WORKSPACE oppia_android_test.bzl
```

After First Review, I will add all bazel files to check
```
./buildifier --lint=warn --mode=check --warnings=all -r app data domain model testing utility BUILD.bazel WORKSPACE oppia_android_test.bzl
```

## Checklist
